src/openms/source/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.cpp

```diff
@@ -266,6 +266,32 @@ namespace OpenMS
     trafo_order = map_sets[last_trafo];
   }
 
+  void MapAlignmentAlgorithmTreeGuided::align(std::vector<FeatureMap>& feature_maps,
+           std::vector<TransformationDescription>& transformations)
+  {
+    // constructing tree
+    vector<vector<double>> maps_ranges(feature_maps.size());  // to save ranges for alignment (larger rt_range -> reference)
+    std::vector<BinaryTreeNode> tree;    // to construct tree with pearson coefficient
+    buildTree(feature_maps, tree, maps_ranges);
+        // print tree
+    ClusterAnalyzer ca;
+    OPENMS_LOG_INFO << "  Alignment follows Newick tree: " << ca.newickTree(tree, true) << endl;
+
+    // alignment
+    vector<Size> trafo_order;
+    FeatureMap map_transformed;
+    vector<FeatureMap> copied_maps = feature_maps;
```

Comment:
put scope (brackets) around copied maps and calculation below? will free the map immidiatly

Reply:
ok